File: introExample/src/main/java/com/wujie/introexample/model/Cat.java
package com.wujie.introexample.model;

import io.realm.RealmObject;
import io.realm.RealmResults;
import io.realm.annotations.LinkingObjects;

/**
 * Created by wujie on 2018/7/17/017.
 */

public class Cat extends RealmObject {
    public String name;
    @LinkingObjects("cats")
    public final RealmResults<Person> owners = null;
}

File: kotlinExample/src/main/java/com/wujie/kotlinexample/model/Cat.kt
package com.wujie.kotlinexample.model

import io.realm.RealmObject
import io.realm.RealmResults
import io.realm.annotations.LinkingObjects

/**
 * Created by wujie on 2018/7/18/018.
 */
class Cat : RealmObject() {
    var name: String? = null

    @LinkingObjects("cats")
    var owners: RealmResults<Person>? = null
}